File: Java/medium/1604. Alert Using Same Key-Card Three or More Times in a One Hour Period/1604_Alert_Using_Same_Key-Card_Three_or_More_Times_in_a_One_Hour_Period.java
import java.util.*;

class Solution {
    public List<String> alertNames(String[] keyName, String[] keyTime) {
        Map<String, List<Integer>> usageMap = new HashMap<>();
        for (int i = 0; i < keyName.length; i++) {
            usageMap.putIfAbsent(keyName[i], new ArrayList<>());
            String[] time = keyTime[i].split(":");
            int minutes = Integer.parseInt(time[0]) * 60 + Integer.parseInt(time[1]);
            usageMap.get(keyName[i]).add(minutes);
        }

        List<String> result = new ArrayList<>();
        for (String name : usageMap.keySet()) {
            List<Integer> times = usageMap.get(name);
            Collections.sort(times);
            for (int i = 2; i < times.size(); i++) {
                if (times.get(i) - times.get(i - 2) <= 60) {
                    result.add(name);
                    break;
                }
            }
        }

        Collections.sort(result);
        return result;
    }
}

File: Golang/1604_Alert_Using_Same_Key-Card_Three_or_More_Times_in_a_One_Hour_Period/1604_Alert_Using_Same_Key-Card_Three_or_More_Times_in_a_One_Hour_Period.go

func alertNames(keyName []string, keyTime []string) []string {
	usageMap := make(map[string][]int)
	for i := 0; i < len(keyName); i++ {
		timeParts := strings.Split(keyTime[i], ":")
		hours, _ := strconv.Atoi(timeParts[0])
		minutes, _ := strconv.Atoi(timeParts[1])
		totalMinutes := hours*60 + minutes
		usageMap[keyName[i]] = append(usageMap[keyName[i]], totalMinutes)
	}

	result := []string{}
	for name, times := range usageMap {
		sort.Ints(times)
		for i := 2; i < len(times); i++ {
			if times[i]-times[i-2] <= 60 {
				result = append(result, name)
				break
			}
		}
	}

	sort.Strings(result)
	return result
}
